addon/mixins/audit-model.js

```diff
@@ -59,17 +59,39 @@ export default Ember.Mixin.create({
   */
   save() {
     let currentDate = new Date();
-    let currentUser = this.get('currentUserName');
-    if (this.get('isNew')) {
-      this.set('createTime', currentDate);
-      this.set('creator', currentUser);
-    }
-
-    if (this.get('hasDirtyAttributes') && !this.get('isDeleted')) {
-      this.set('editTime', currentDate);
-      this.set('editor', currentUser);
-    }
-
-    return this._super(...arguments);
+    let superFunc = this._super;
+    let _this = this;
+
+    let currentUserPromise = new Ember.RSVP.Promise((resolve, reject) => {
+      let userName = this.get('currentUserName');
+      if (userName instanceof Ember.RSVP.Promise) {
+        userName.then(name => {
+          resolve(name);
+        }).catch(() => {
```

Comment:
Need to add `reason` parameter to anonymous function and pass it to `reject` call.